public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -1,339 +1,63 @@
-                  tooltip="If you are assuming a role in another account, that has been created with an external ID, specify the external ID here."
-                >
-                  External ID
-                </InlineFormLabel>
-                <div className="width-30">
-                  <Input
-                    className="width-30"
-                    placeholder="External ID"
-                    value={options.jsonData.externalId || ''}
-                    onChange={onUpdateDatasourceJsonDataOption(this.props, 'externalId')}
-                  />
-                </div>
-              </div>
-            </div>
-          </div>
-          <div className="gf-form-inline">
-            <div className="gf-form">
-              <InlineFormLabel
-                className="width-14"
-                tooltip="Specify the region, such as for US West (Oregon) use ` us-west-2 ` as the region."
-              >
-                Default Region
-              </InlineFormLabel>
-              <Select
-                className="width-30"
-                value={regions.find((region) => region.value === options.jsonData.defaultRegion)}
-                options={regions}
-                defaultValue={options.jsonData.defaultRegion}
-                onChange={onUpdateDatasourceJsonDataOptionSelect(this.props, 'defaultRegion')}
-              />
-            </div>
-          </div>
-          <div className="gf-form-inline">
-            <div className="gf-form">
-              <InlineFormLabel className="width-14" tooltip="Namespaces of Custom Metrics.">
-                Custom Metrics
-              </InlineFormLabel>
-              <Input
-                className="width-30"
-                placeholder="Namespace1,Namespace2"
-                value={options.jsonData.customMetricsNamespaces || ''}
-                onChange={onUpdateDatasourceJsonDataOption(this.props, 'customMetricsNamespaces')}
-              />
-            </div>
-          </div>
-          <div className="gf-form-inline">
-            <div className="gf-form">
-              <InlineFormLabel className="width-14" tooltip="Optionally, specify a custom endpoint for the service.">
-                Endpoint
-              </InlineFormLabel>
-              <div className="width-30">
-                <Input
-                  className="width-30"
-                  placeholder={'https://{service}.{region}.amazonaws.com'}
-                  value={options.jsonData.endpoint || ''}
-                  onChange={onUpdateDatasourceJsonDataOption(this.props, 'endpoint')}
-                />
-              </div>
-            </div>
-          </div>
-        </div>
-      </>
-    );
-  }
-}
-
-export default ConfigEditor;
+  }, []);
+
+  return (
+    <>
+      <ConnectionConfig
+        {...props}
+        loadRegions={
+          datasource &&
+          (() => datasource!.getRegions().then((r) => r.filter((r) => r.value !== 'default').map((v) => v.value)))
```

Comment:
As `datasource` may be undefined on first add this could fail.

Reply:
The loadRegions prop in ConnectionConfig is nullable so it should be fine also before the datasource is saved.